pkg/state/state_ofed.go

```diff
@@ -87,9 +110,76 @@ type ofedRuntimeSpec struct {
 }
 
 type ofedManifestRenderData struct {
-	CrSpec       *mellanoxv1alpha1.OFEDDriverSpec
-	NodeAffinity *v1.NodeAffinity
-	RuntimeSpec  *ofedRuntimeSpec
+	CrSpec                 *mellanoxv1alpha1.OFEDDriverSpec
+	NodeAffinity           *v1.NodeAffinity
+	RuntimeSpec            *ofedRuntimeSpec
+	AdditionalVolumeMounts additionalVolumeMounts
+}
+
+// getCertConfigPath returns the standard OS specific path for ssl keys/certificates
+func getCertConfigPath(osname string) (string, error) {
+	if path, ok := CertConfigPathMap[osname]; ok {
+		return path, nil
+	}
+	return "", fmt.Errorf("distribution not supported")
+}
+
+// getRepoConfigPath returns the standard OS specific path for repository configuration files
+func getRepoConfigPath(osname string) (string, error) {
+	if path, ok := RepoConfigPathMap[osname]; ok {
+		return path, nil
+	}
+	return "", fmt.Errorf("distribution not supported")
+}
+
+// createConfigMapVolumeMounts creates a VolumeMount for each key
+// in the ConfigMap. Use subPath to ensure original contents
+// at destinationDir are not overwritten.
+func (s *stateOFED) createConfigMapVolumeMounts(namespace, configMapName, destinationDir string) (
+	[]v1.VolumeMount, []v1.KeyToPath, error) {
+	// get the ConfigMap
+	cm := &v1.ConfigMap{}
+
+	objKey := client.ObjectKey{Namespace: namespace, Name: configMapName}
+	err := s.client.Get(context.TODO(), objKey, cm)
+	if err != nil {
+		return nil, nil, fmt.Errorf("ERROR: could not get ConfigMap %s from client: %v", configMapName, err)
+	}
+
+	// create one volume mount per file in the ConfigMap and use subPath
+	var filenames = make([]string, 0, len(cm.Data))
+	for filename := range cm.Data {
+		filenames = append(filenames, filename)
+	}
+	// sort so volume mounts are added to spec in deterministic order to make testing easier
+	sort.Strings(filenames)
+	var itemsToInclude = make([]v1.KeyToPath, 0, len(filenames))
+	var volumeMounts = make([]v1.VolumeMount, 0, len(filenames))
+	for _, filename := range filenames {
+		volumeMounts = append(volumeMounts,
+			v1.VolumeMount{
+				Name:      configMapName,
+				ReadOnly:  true,
+				MountPath: filepath.Join(destinationDir, filename),
+				SubPath:   filename})
+		itemsToInclude = append(itemsToInclude, v1.KeyToPath{
+			Key:  filename,
+			Path: filename,
+		})
+	}
+	return volumeMounts, itemsToInclude, nil
+}
+
+func (s *stateOFED) createConfigMapVolume(configMapName string, itemsToInclude []v1.KeyToPath) v1.Volume {
```

Comment:
question: doesnt it make more sense to move the logic of fetching config map and creating "itemsToInclude" here ? then using it in createConfigMapVolumeMounts

i.e 
you first create the volume and then create the mounts for the volume.

then `createConfigMapVolumeMounts` would be just `createVolumeMounts(volumeName string, destDir string, items []v1.KeyToPath)`

Reply:
it's already wrapped into mountAdditionalVolumesFromConfigMap function witch  should be used by the controller